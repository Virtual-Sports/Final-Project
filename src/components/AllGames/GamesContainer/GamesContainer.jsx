import React, { useState } from 'react'
import PropTypes from 'prop-types'

import styles from './GamesContainer.module.css'

import GameCard from '../GameCard/GameCard'
import { ReactComponent as PMSorry } from '../../../resources/icons/pm-sorry.svg'
import { gamesContainer as messages } from '../../../shared/messages'

import { Collapse } from 'react-collapse'

GamesContainer.propTypes = {
    title: PropTypes.string.isRequired,
    games: PropTypes.array,
    search: PropTypes.bool,
}

function GamesContainer({ title = '', games = [], search }) {
    const [isOpened, setIsOpened] = useState(false)

    return (
        <>
            <Collapse isOpened={isOpened} duration={5000}>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
                <div>Random content</div>
            </Collapse>
            {games.length ? (
                <div className={styles['container']}>
                    <div className={styles['header']}>
                        <h2
                            className={styles['title']}
                            onClick={() => {
                                setIsOpened(!isOpened)
                            }}
                        >
                            {title}&nbsp;
                            <span className={styles['count']}>
                                [{games.length}]
                            </span>
                        </h2>
                    </div>
                    <div className={styles['games']}>
                        {games.map(game => (
                            <GameCard
                                key={game.id}
                                id={game.id}
                                title={game.displayName}
                                img={game.image}
                            />
                        ))}
                    </div>
                </div>
            ) : (
                search && (
                    <div className={styles['nothing-found']}>
                        <span className={styles['message']}>
                            {messages.nothingFound}
                        </span>
                        <PMSorry className={styles['image']} />
                    </div>
                )
            )}
        </>
    )
}

export default GamesContainer
