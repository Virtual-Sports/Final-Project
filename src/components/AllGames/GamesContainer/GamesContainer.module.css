.container {
    width: 100%;
    margin: 15px 0;
}

.header {
    font-weight: 700;
    font-size: 0.9em;
    display: flex;
    align-items: center;
    margin-bottom: 15px;
    padding-bottom: 10px;
    border-bottom-width: 1px;
    border-bottom-style: solid;
    border-image: linear-gradient(
        90deg,
        rgba(252, 248, 84, 1) 0%,
        rgba(253, 251, 168, 0.4640231092436975) 50%,
        rgba(255, 255, 255, 0) 100%
    );
    border-image-slice: 1;
}

.header img {
    height: 18px;
    width: 18px;
    margin-right: 10px;
    filter: invert(100%) sepia(3%) saturate(13%) hue-rotate(81deg)
        brightness(106%) contrast(106%);
}

.games {
    display: flex;
    align-items: center;
    justify-content: flex-start;
    flex-wrap: wrap;
}

.count {
    color: #fcf854;
    font-size: 0.75em;
}

.nothing-found {
    display: flex;
    align-items: center;
    flex-direction: row-reverse;
    justify-content: center;
}

.message {
    margin-bottom: 1em;
    white-space: pre-line;
    text-align: center;
}

.image {
    width: 350px;
}
