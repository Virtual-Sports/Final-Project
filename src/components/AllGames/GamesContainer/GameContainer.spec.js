import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import GamesContainer from './GamesContainer.jsx'
import { ReactComponent as PMSorry } from '../../../resources/icons/pm-sorry.svg'

describe('GamesContainer component', () => {
    it('should render without crashing', () => {
        shallow(<GamesContainer title={'title'} games={[]} />)
    })

    it('should contain sorry image if no games passed', () => {
        const wrapper = shallow(<GamesContainer title={'title'} games={[]} />)

        expect(wrapper.containsMatchingElement(<PMSorry />)).toEqual(true)
    })

    it('should display default image if no img prop given', () => {
        const games = [
            {
                id: 0,
            },
            {
                id: 1,
            },
            {
                id: 2,
            },
        ]

        const tree = renderer
            .create(
                <BrowserRouter>
                    <GamesContainer title={'title'} games={games} />
                </BrowserRouter>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
