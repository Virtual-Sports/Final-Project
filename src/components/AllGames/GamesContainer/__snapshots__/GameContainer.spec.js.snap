// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard component should display default image if no img prop given 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <h2
      className="title"
    >
      title
       
      <span
        className="count"
      >
        [
        3
        ]
      </span>
    </h2>
  </div>
  <div
    className="games"
  >
    <a
      className="card"
      href="/game/0"
      onClick={[Function]}
    >
      <img
        onError={[Function]}
        src="https://via.placeholder.com/250x250"
      />
      <p
        className="description"
      >
        No title
      </p>
    </a>
    <a
      className="card"
      href="/game/1"
      onClick={[Function]}
    >
      <img
        onError={[Function]}
        src="https://via.placeholder.com/250x250"
      />
      <p
        className="description"
      >
        No title
      </p>
    </a>
    <a
      className="card"
      href="/game/2"
      onClick={[Function]}
    >
      <img
        onError={[Function]}
        src="https://via.placeholder.com/250x250"
      />
      <p
        className="description"
      >
        No title
      </p>
    </a>
  </div>
</div>
`;

exports[`GamesContainer component should display default image if no img prop given 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <h2
      className="title"
    >
      title
       
      <span
        className="count"
      >
        [
        3
        ]
      </span>
    </h2>
  </div>
  <div
    className="games"
  >
    <a
      className="card"
      href="/game/0"
      onClick={[Function]}
    >
      <img
        onError={[Function]}
        src="https://via.placeholder.com/250x250"
      />
      <p
        className="description"
      >
        No title
      </p>
    </a>
    <a
      className="card"
      href="/game/1"
      onClick={[Function]}
    >
      <img
        onError={[Function]}
        src="https://via.placeholder.com/250x250"
      />
      <p
        className="description"
      >
        No title
      </p>
    </a>
    <a
      className="card"
      href="/game/2"
      onClick={[Function]}
    >
      <img
        onError={[Function]}
        src="https://via.placeholder.com/250x250"
      />
      <p
        className="description"
      >
        No title
      </p>
    </a>
  </div>
</div>
`;
